runtime/pkg/observability/middleware.go

```diff
@@ -62,16 +64,47 @@ func LoggingUnaryServerInterceptor(logger *zap.Logger) grpc.UnaryServerIntercept
 // LoggingStreamServerInterceptor is the streaming equivalent of LoggingUnaryServerInterceptor
 func LoggingStreamServerInterceptor(logger *zap.Logger) grpc.StreamServerInterceptor {
 	return logging.StreamServerInterceptor(
-		grpczap.InterceptorLogger(logger),
-		logging.WithDecider(logFinishDecider),
+		zapInterceptorLogger(logger),
+		logging.WithLogOnEvents(logging.StartCall, logging.FinishCall),
+		logging.WithFieldsFromContext(tracingFieldsFromCtx),
 		logging.WithCodes(errorToCode),
 		logging.WithLevels(grpcCodeToLevel),
 	)
 }
 
-// logFinishDecider filters which calls to log. It logs all calls (start and finish).
-func logFinishDecider(fullMethodName string, err error) logging.Decision {
-	return logging.LogStartAndFinishCall
+// zapInterceptorLogger adapts zap logger to a gRPC interceptor logger.
+// Source: https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/interceptors/logging/examples/zap/example_test.go
+func zapInterceptorLogger(l *zap.Logger) logging.Logger {
+	return logging.LoggerFunc(func(ctx context.Context, lvl logging.Level, msg string, fields ...any) {
+		f := make([]zap.Field, 0, len(fields)/2)
+		i := logging.Fields(fields).Iterator()
+		for i.Next() {
+			k, v := i.At()
+			f = append(f, zap.Any(k, v))
+		}
+
+		switch lvl {
+		case logging.LevelDebug:
+			l.Debug(msg, f...)
+		case logging.LevelInfo:
+			l.Info(msg, f...)
+		case logging.LevelWarn:
+			l.Warn(msg, f...)
+		case logging.LevelError:
+			l.Error(msg, f...)
+		default:
+			panic(fmt.Sprintf("unknown level %v", lvl))
+		}
+	})
+}
+
+// tracingFieldsFromCtx picks tracing-related fields from the ctx for use in the gRPC logging interceptor.
+func tracingFieldsFromCtx(ctx context.Context) logging.Fields {
+	sctx := trace.SpanFromContext(ctx).SpanContext()
+	if !sctx.IsValid() {
+		return nil
+	}
+	return []any{"trace_id", sctx.TraceID().String()}
```

Comment:
Do we also want `span_id` ?

Reply:
I also initially thought so, but the examples for otel grpc did not have it (only `trace_id`). @kaspersjo, what do you think?